cpp/tests/copying/concatenate_tests.cu

```diff
@@ -335,103 +336,430 @@ TEST_F(TableTest, ConcatenateTablesWithOffsetsAndNulls)
+    thrust::fill(rmm::exec_policy(),
+                 offsets->mutable_view().begin<offset_type>(),
+                 offsets->mutable_view().end<offset_type>(),
+                 string_size);
+    thrust::exclusive_scan(rmm::exec_policy(),
+                           offsets->view().begin<offset_type>(),
+                           offsets->view().end<offset_type>(),
+                           offsets->mutable_view().begin<offset_type>());
+    auto many_chars = cudf::make_fixed_width_column(data_type{type_id::INT8}, inner_size);
+    auto col        = cudf::make_strings_column(
+      num_rows, std::move(offsets), std::move(many_chars), 0, rmm::device_buffer{});
+
+    auto sliced = cudf::slice(*col, {16, 32});
+
+    // 192 outer rows
+    // 201,326,592 inner rows
+    table_view a({sliced[0]});
+    cudf::concatenate(std::vector<table_view>({a, a, a, a, a, a, a, a, a, a, a, a}));
+  }
+
+  // list<int8> column
+  {
+    constexpr size_type inner_size =
+      static_cast<size_type>(static_cast<uint32_t>(1024) * 1024 * 1024);
+    constexpr size_type num_rows  = 1024;
+    constexpr size_type list_size = inner_size / num_rows;
+
+    auto offsets = cudf::make_fixed_width_column(data_type{type_id::INT32}, num_rows + 1);
+    thrust::fill(rmm::exec_policy(),
+                 offsets->mutable_view().begin<offset_type>(),
+                 offsets->mutable_view().end<offset_type>(),
+                 list_size);
+    thrust::exclusive_scan(rmm::exec_policy(),
+                           offsets->view().begin<offset_type>(),
+                           offsets->view().end<offset_type>(),
+                           offsets->mutable_view().begin<offset_type>());
+    auto many_chars = cudf::make_fixed_width_column(data_type{type_id::INT8}, inner_size);
+    auto col        = cudf::make_lists_column(
+      num_rows, std::move(offsets), std::move(many_chars), 0, rmm::device_buffer{});
+
+    auto sliced = cudf::slice(*col, {16, 32});
+
+    // 192 outer rows
+    // 201,326,592 inner rows
+    table_view a({sliced[0]});
+    cudf::concatenate(std::vector<table_view>({a, a, a, a, a, a, a, a, a, a, a, a}));
+  }
+
+  // struct<int8, list>
+  {
+    constexpr size_type inner_size =
+      static_cast<size_type>(static_cast<uint32_t>(1024) * 1024 * 1024);
+    constexpr size_type num_rows  = 1024;
+    constexpr size_type list_size = inner_size / num_rows;
+
+    auto offsets = cudf::make_fixed_width_column(data_type{type_id::INT32}, num_rows + 1);
+    thrust::fill(rmm::exec_policy(),
+                 offsets->mutable_view().begin<offset_type>(),
+                 offsets->mutable_view().end<offset_type>(),
+                 list_size);
+    thrust::exclusive_scan(rmm::exec_policy(),
+                           offsets->view().begin<offset_type>(),
+                           offsets->view().end<offset_type>(),
+                           offsets->mutable_view().begin<offset_type>());
```

Comment:
You could use `std::fill` and `std::inclusive_scan` to fill out a host memory buffer and then `cudaMemcpy` to the offsets column in order to avoid changing this to ` .cu` file.

Reply:
The numbers in some of these cases are pretty big.  I switched to thrust for speed purposes.